coverage/src/main/java/gov/cms/ab2d/coverage/repository/CoverageServiceRepository.java

```diff
@@ -95,6 +96,13 @@ public class CoverageServiceRepository {
     private static final String SELECT_DISTINCT_COVERAGE_BY_PERIOD_COUNT = "SELECT COUNT(DISTINCT beneficiary_id) FROM coverage" +
             " WHERE bene_coverage_period_id IN(:ids) AND contract = :contract AND year IN (:years)";
 
+    /**
+     * Return a count of all beneficiaries who aggred to share their data associated with an {@link CoveragePeriod}
+     * from any event. For those beneficiaries out_out_flag equals false in the public.coverage table.
+     */
+    private static final String SELECT_DISTINCT_OPTOUT_COVERAGE_BY_PERIOD_COUNT = "SELECT COUNT(DISTINCT beneficiary_id) FROM coverage" +
```

Comment:
Not sure why any of these queries (both  old existing queries and new queries) consider month in them as a parameter .. lets say  if this query is run in October and the coverage goes back only 3 months. why grab all unnecessary data for whole year .. I see the distinct but probably if we can consider the current date and decide to go back 3 month will it be more efficient.. thoughts.. This shudnt stop PR from merging.. May be a spike after?

Reply:
[AB2D-5736](https://jira.cms.gov/browse/AB2D-5736) Spike: Use current date in SELECT_DISTINCT_COVERAGE_BY_PERIOD_COUNT statement